tools/wptrunner/wptrunner/executors/executorcontentshell.py

```diff
@@ -186,6 +191,9 @@ def _convert_exception(test, exception, errors):
         return (test.result_cls("EXTERNAL-TIMEOUT", errors), [])
     if isinstance(exception, CrashError):
         return (test.result_cls("CRASH", errors), [])
+    if isinstance(exception, LeakError):
+        test.restart_test_after = True
```

Comment:
I noticed that once `restart_test_after=True` is set, it becomes set permanently for the lifetime of `test`. It seems that a leak in an earlier `--rerun` (i.e., `run_wpt_tests.py --repeat-each`) will force restarts in later ones, which seems undesirable.

I think it would be better to communicate the leakage through the test result directly instead of using `test` as a mutable container, which it was not meant to be.